packages/artifact/__tests__/util.test.ts

```diff
@@ -59,3 +61,106 @@ describe('get-backend-ids-from-token', () => {
     )
   })
 })
+
+jest.mock('@actions/core')
+
+describe('maskSigUrl', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('does nothing if no sig parameter is present', () => {
+    const url = 'https://example.com'
+    maskSigUrl(url)
+    expect(setSecret).not.toHaveBeenCalled()
+  })
+
+  it('masks the sig parameter in the middle of the URL and sets it as a secret', () => {
+    const url = 'https://example.com/?param1=value1&sig=12345&param2=value2'
+    maskSigUrl(url)
+    expect(setSecret).toHaveBeenCalledWith('12345')
+    expect(setSecret).toHaveBeenCalledWith(encodeURIComponent('12345'))
```

Comment:
how about having the test signature be a value that will be encode differently?

like `foo bar` and `foo%20bar`

Reply:
see that's why I think we don't need it, because I don't think we will get a URL that it not encoded properly, though happy to hear otherwise if that could be the case?

Reply:
i stand corrected, will update it

Reply:
added more